### PR TITLE
[141, 142, 143, 144, irods/irods_5669, irods/irods_6086] Build spdlog, catch2, and cppzmq with cmake

### PR DESCRIPTION
**Pull request by SwooshyCueb**
This PR accomplishes a few things:
- Puts CMake configuration files in the spdlog, catch2, and cppzmq packages
- Puts catch2's headers in the right place
- Removes the extra copy of fmt from spdlog

### CODE REVIEW

File: versions.json [hunk 1]
```diff
@@ -411,16 +413,17 @@
         "commitish": "v1.9.2",
         "version_string": "1.9.2",
         "license": "MIT",
-        "consortium_build_number": "0",
+        "consortium_build_number": "2",
```
Comment (trel) [inline versions.json:416]: why jumping to 2?

Comment (SwooshyCueb) [reply versions.json:416]: incremented twice in two separate commits

Comment (trel) [reply versions.json:416]: ah, i see - but we haven't actually pushed/published since 0 - so I think we can just go to 1 for now?

**Comment by trel** [comment]
rest looks good - # when ready

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**